db/src/db.rs

```diff
@@ -416,45 +437,438 @@ impl DB {
+            // TODO: compress bit flags into a single bit field, and expand when inserting into
+            // `datoms` and `transactions`.
+            // TODO: drop tx0 entirely.
+            r#"CREATE TABLE temp.exact_searches (
+               e0 INTEGER NOT NULL,
+               a0 SMALLINT NOT NULL,
+               v0 BLOB NOT NULL,
+               value_type_tag0 SMALLINT NOT NULL,
+               tx0 INTEGER NOT NULL,
+               added0 TINYINT NOT NULL,
+               index_avet0 TINYINT NOT NULL,
+               index_vaet0 TINYINT NOT NULL,
+               index_fulltext0 TINYINT NOT NULL,
+               unique_value0 TINYINT NOT NULL)"#,
+            // There's no real need to split exact and inexact searches, so long as we keep things
+            // in the correct place and performant.  Splitting has the advantage of being explicit
+            // and slightly easier to read, so we'll do that to start.
+            r#"DROP TABLE IF EXISTS temp.inexact_searches"#,
+            r#"CREATE TABLE temp.inexact_searches (
+               e0 INTEGER NOT NULL,
+               a0 SMALLINT NOT NULL,
+               v0 BLOB NOT NULL,
+               value_type_tag0 SMALLINT NOT NULL,
+               tx0 INTEGER NOT NULL,
+               added0 TINYINT NOT NULL,
+               index_avet0 TINYINT NOT NULL,
+               index_vaet0 TINYINT NOT NULL,
+               index_fulltext0 TINYINT NOT NULL,
+               unique_value0 TINYINT NOT NULL)"#,
+            r#"DROP TABLE IF EXISTS temp.search_results"#,
+            // TODO: don't encode search_type as a STRING.  This is explicit and much easier to read
+            // than another flag, so we'll do it to start, and optimize later.
+            r#"CREATE TABLE temp.search_results (
+               e0 INTEGER NOT NULL,
+               a0 SMALLINT NOT NULL,
+               v0 BLOB NOT NULL,
+               value_type_tag0 SMALLINT NOT NULL,
+               tx0 INTEGER NOT NULL,
+               added0 TINYINT NOT NULL,
+               index_avet0 TINYINT NOT NULL,
+               index_vaet0 TINYINT NOT NULL,
+               index_fulltext0 TINYINT NOT NULL,
+               unique_value0 TINYINT NOT NULL,
+               search_type STRING NOT NULL,
+               rid INTEGER,
+               v BLOB)"#,
+            // It is an error to transact the same [e a v] twice in one transaction.  This index will
+            // cause insertion to fail if a transaction tries to do that.  (Sadly, the failure is
+            // opaque.)
+            //
+            // N.b.: temp goes on index name, not table name.  See http://stackoverflow.com/a/22308016.
+            r#"CREATE UNIQUE INDEX IF NOT EXISTS temp.search_results_unique ON search_results (e0, a0, v0, value_type_tag0)"#,
+        ];
+
+        for statement in &statements {
+            let mut stmt = conn.prepare_cached(statement)?;
+            stmt.execute(&[])
+                .map(|_c| ())
```

Comment:
Yeah, that or `.success()`.  I don't really care to define this right now; it could be a good first bug.